scraper/_utils.py

```diff
@@ -0,0 +1,28 @@
+import json
+from elasticsearch import exceptions
+
+
+def insert_to_es(es, s, name):
+    r = {}
+    try:
+        r['title'] = s.title()
+        r['ingredients'] = s.ingredients()
+        r['link'] = s.url
+        r['image'] = s.image()
+        r['total_time'] = s.total_time()
+        r['tags'] = s.tags()
+        r['diet'] = s.suitable_for_diet()
+        r['id'] = name + s.id()
+    except Exception as e:
+        print(e)
+        print('failed to get ' + s.url)
+    try:
+        r['rating'] = s.ratings()
+    except (NotImplementedError, TypeError) as e:
+        r['rating'] = None
+    try:
+        s = es.create(index='recipes', doc_type='recipe', id=r['id'], body=json.dumps(r))
```

Comment:
It should be better for run time if you use the bulk api to add multiple recipes at once.
When you start experimenting with long crawls I suggest you check it out.
https://elasticsearch-py.readthedocs.io/en/master/helpers.html#example

Reply:
I will look into that, the issue I have is that I have to keep track of how many 409's (i.e - how many recipes have already been added before) I receive in order to do the page-switching correctly

Reply:
btw, doc_type is already deprecated in ES 7.x, each index has only one type so no need for that